modules/product/src/main/java/com/opengamma/strata/product/GenericSecurityTrade.java

```diff
@@ -23,28 +23,17 @@
 import org.joda.beans.impl.direct.DirectMetaProperty;
 import org.joda.beans.impl.direct.DirectMetaPropertyMap;
 
-import com.opengamma.strata.collect.id.LinkResolver;
-import com.opengamma.strata.collect.id.StandardId;
-import com.opengamma.strata.product.SecurityLink;
-import com.opengamma.strata.product.SecurityTrade;
-import com.opengamma.strata.product.TradeInfo;
-
 /**
- * A trade in a generic futures contract based on an expiry month.
- * <p>
- * A trade in an underlying {@link GenericFuture}.
+ * A trade representing the purchase or sale of a security,
+ * where the security is embedded ready for mark-to-market pricing.
  * <p>
- * A future is a financial instrument that is based on the future value of an underlying.
- * The buyer is typically required to purchase the underlying at a price fixed in advance.
- * This class represents the structure of a single futures contract.
- * <p>
- * For example, an airline can use a future to lock in the price of jet fuel in advance.
- * The airline will lose on the trade if jet fuel drops in price, but gain on the trade
- * if jet fuel rises in price.
+ * This trade represents a trade in a security, defined by a quantity and price.
+ * The security is embedded directly, however the underlying product model is not available.
+ * The security may be of any kind, including equities, bonds and exchange traded derivative (ETD).
  */
-@BeanDefinition
-public final class GenericFutureTrade
-    implements SecurityTrade<GenericFuture>, ImmutableBean, Serializable {
+@BeanDefinition(constructorScope = "package")
+public final class GenericSecurityTrade
```

Comment:
What's this class for? It seems to live in the middle ground between securities, where other securities are linked by ID, and products, where other products are directly referenced.